packages/chevrotain/src/parse/grammar/checks.ts

```diff
@@ -432,34 +429,30 @@ export function validateAmbiguousAlternationAlternatives(
   // - https://github.com/chevrotain/chevrotain/issues/869
   ors = reject(ors, (currOr) => currOr.ignoreAmbiguities === true)
 
-  const errors = reduce(
-    ors,
-    (result, currOr: Alternation) => {
-      const currOccurrence = currOr.idx
-      const actualMaxLookahead = currOr.maxLookahead || globalMaxLookahead
-      const alternatives = getLookaheadPathsForOr(
-        currOccurrence,
-        topLevelRule,
-        actualMaxLookahead,
-        currOr
-      )
-      const altsAmbiguityErrors = checkAlternativesAmbiguities(
-        alternatives,
-        currOr,
-        topLevelRule,
-        errMsgProvider
-      )
-      const altsPrefixAmbiguityErrors = checkPrefixAlternativesAmbiguities(
-        alternatives,
-        currOr,
-        topLevelRule,
-        errMsgProvider
-      )
+  const errors = flatMap(ors, (currOr: Alternation) => {
```

Comment:
this usage of `flatMap` seems better than the previous ones as there is not branch that returns an empty array (explicitly).

Was there an error in this code due to `strictNullChecks` ?

Reply:
Yeah iirc the initAccumulator would have require an explicit type with reduce, so flatMap is nicer here. And returning an empty array above makes  things uniform.